vault_program/src/mint_to.rs

```diff
@@ -93,6 +114,21 @@ pub fn process_mint(
         )?;
     }
 
+    let vault_amount_after = {
```

Comment:
imo worth moving the checks to below the transfer + mint invokes. then do:

assert(vault.vrt_supply == vrt_mint.total_supply);
assert(vault.tokens_deposited == token_account.amount);

Reply:
ah yeah that makes sense, good call